### PR TITLE
Add Pre-commit hooks

### LINKED ISSUE #22: Add pre-commit hooks
Desired features of pre-commit hooks for geocat-applications:
- notebook checking
- autoupdate
- allow single quotes
- ability to add exceptions on a case-by-case basis (I believe uxarray did something like this)

Other considerations:
- also look into what uxarray did with ruff for notebooks?

### PR DESCRIPTION
**Pull request by cyschneck**
## PR Summary
Add pre-commit hooks

## Related Tickets & Documents
Closes #22

## PR Checklist
**General**
- [x] PR includes a summary of changes
- [x] Link relevant issues, make one if none exist

### CODE REVIEW

**Comment by cyschneck** [comment]
The `ruff.toml` is currently set up to ignore `E402 Module level import not at top of cell` for the Jupyter Notebooks and to preserve the ability to use single/double quotes.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - id: codespell
+        additional_dependencies:
+          - tomli
+
+-   repo: https://github.com/pre-commit/pre-commit-hooks    # Some common pre-commit hooks
+    rev: v4.6.0
+    hooks:
+    - id: check-yaml               # Checks the syntax of .yaml files.
+      args: [--allow-multiple-documents]
+      exclude: 'meta.yaml'    # Exclude this because it gives an error for '%' in Line 1 and couldn't fix yet
```
Comment (anissa111) [inline .pre-commit-config.yaml:35]:
```suggestion
```
We shouldn't need this here

**Review by anissa111 — COMMENTED** [review]
I think it's nearly there, just a few small points

File: ruff.toml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[format]
+# preserve mixture of single and double quotes
+quote-style = "preserve"
+
+[per-file-ignores]
+# E402 Module level import not at top of cell
+"applications/data_analysis/sample.ipynb" = ["E402"]
+"applications/file_io/sample.ipynb" = ["E402"]
+"applications/plot_types/contour.ipynb" = ["E402"]
```
[the remark this answers is not in the document]
Comment (cyschneck) [reply ruff.toml]: Sure thing, I will set it to ignore the rule

**Review by anissa111 — APPROVED** [review]

**Pull request merged**